### PR TITLE
Feature/java 11 build

### PR DESCRIPTION
**Pull request by whyicantusemyemailasusername**
Various fixes to make maven build work on java 11 (see #497, #498 ) :

* lombok version update. it is used in single file (test); can be removed completely from the project
* add explicit jaxb dependencies where required; see https://stackoverflow.com/questions/43574426/how-to-resolve-java-lang-noclassdeffounderror-javax-xml-bind-jaxbexception-in-j
* update mockito versions
* update maven plugin versions

### CODE REVIEW

**Comment by whyicantusemyemailasusername** [comment]
- added java9+ profiles for jaxb dependencies
- removed java8 profile for doclint as pippo requires java8. also selector `[1.8,)` doesn't work for 9+ jdks.

File: pom.xml [hunk 1]
```diff
@@ -176,15 +188,6 @@
     </modules>
 
     <profiles>
-        <profile>
```
Comment (decebals) [inline pom.xml]: Do yoy think that we must delete this profile (jdk8) or it's a mistake? The require version for PF4J is java 7.

Comment (whyicantusemyemailasusername) [reply pom.xml]: Pippo requires 1.8+: `<java.version>1.8</java.version>` The profile was always active (actually should be active but it was not on 9), and the parameter is explicitly set in plugin configuration

Comment (decebals) [reply pom.xml]: My mistake, sorry. I am working on a feature on PF4J (a project initiated by me that requires Java 1.7) and from this reason I mix things.

**Comment by decebals** [comment]
Thanks!

**Comment by decebals** [comment]
Maybe we must update travis CI config file to add openjdk 11 (see https://github.com/pf4j/pf4j/blob/master/.travis.yml).

**Comment by whyicantusemyemailasusername** [comment]
> Maybe we must update travis CI config file to add openjdk 11 (see https://github.com/pf4j/pf4j/blob/master/.travis.yml).

see #500

**Pull request merged**